drivers/usb/dwc3/core.c

```diff
@@ -282,6 +314,11 @@ int dwc3_core_soft_reset(struct dwc3 *dwc)
 	if (dwc->current_dr_role == DWC3_GCTL_PRTCAP_HOST)
 		return 0;
 
+	if (dwc->has_dsm_for_softreset) {
+		dev_dbg(dwc->dev, "calling dwc3_pci_dsm_soft_reset()");
+		return dwc3_pci_dsm_soft_reset(dwc->dev);
+	}
+
```

Comment:
This is a resolution of a nontrivial conflict; I'm uncertain of whether this conditional should go before or after the following one.

Also, @gratian mentioned that we might want to look into dropping this patch altogether:

> For the dwc3 softreset thing I think I would prefer to drop our patch if we think the upstream code is a replacement. That patch has always been iffy because we got it via a OE beta drop (tar.gz OE sources) with the early access silicon from Intel and then it never showed up upstream or in OE. I need to take a closer look but it sure would be nice to get rid of our patch

Reply:
@xhuff did this merge differently in #139 and in [comparing the two](https://github.com/ni/linux/pull/139#discussion_r1376976821) I think the `if (dwc->has_dsm_for_softreset)` conditional should go after the now-split `dwc->dr_mode == USB_DR_MODE_HOST` || `dwc->current_dr_role == DWC3_GCTL_PRTCAP_HOST` cases.

But yeah, it's a weird patch, and I don't know if we still need to be carrying it.

Reply:
OK, since this is the "next" kernel and the point of having it is to allow for testing/trying things like this my proposal is to drop our out-of-tree patch and schedule a follow up test before the next major upgrade.
@xhuff @bstreiff Thoughts?

Reply:
Seems reasonable to me, and appropriate for the purpose of 'next'! We'll get some measure of test coverage by our key internal client kernel-next testing in the meantime, too.

Reply:
I like this proposal, let's do it. We can add the patch back if it turns out we need it.